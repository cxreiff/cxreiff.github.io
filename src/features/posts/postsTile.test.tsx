import { render, fireEvent } from '@testing-library/preact'

import PostsTile from './postsTile.component'

const mockNavigate = jest.fn()
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
}))

const renderComponent = () => render(
    <PostsTile
        id={'test'}
        post={{
            content: 'test',
            data: {
                key: 'test',
                title: 'test',
                description: 'test',
                date: '2020-01-01',
            }
        }}
    />
)

test('should render', () => {
    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
})

test('should navigate to post content', () => {
    const { getByRole } = renderComponent()
    
    fireEvent.click(getByRole('article'))

    expect(mockNavigate).toHaveBeenCalled()
})
