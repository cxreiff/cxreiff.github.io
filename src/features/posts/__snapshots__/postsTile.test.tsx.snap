// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <article
    class="card card_clickable"
  >
    <header>
      <h4>
        test
      </h4>
    </header>
    <footer>
      Invalid Date
    </footer>
    <p>
      test
    </p>
  </article>
</div>
`;
